frontend/src/components/dateSelector.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from "react";
+
+export interface DateSelectorProps {
+  onChange: (value: Date) => void;
+  defaultDate: string | null | undefined;
+}
+
+const DateSelector: React.FC<DateSelectorProps> = ({
+  onChange,
+  defaultDate,
+}) => {
+  const [releaseDate, setReleaseDate] = useState<Date | null>(
+    defaultDate ? new Date(defaultDate) : null,
+  );
+  const [releaseTime, setReleaseTime] = useState<Date | null>(
+    defaultDate ? new Date(defaultDate) : null,
+  );
+
+  const [userChangedDate, setUserChangedDate] = useState<boolean>(false);
+  useEffect(() => {
+    if (userChangedDate && releaseDate && releaseTime) {
+      let newReleaseDateTime = releaseDate;
+      newReleaseDateTime.setHours(releaseTime.getUTCHours());
+      newReleaseDateTime.setMinutes(releaseTime.getUTCMinutes());
+      newReleaseDateTime.setSeconds(releaseTime.getUTCSeconds());
+      onChange(newReleaseDateTime);
+      console.log(newReleaseDateTime);
+    }
+  }, [releaseDate, releaseTime]);
+
+  //This is absolutely vile and cursed, but i honestly cannot look at dates again wihtout going absolutely apeshit
+  //Besides, it only shows it to the user :)
```

Comment:
Could we have a more reasonable comment, explaining what is the problem and why?
To instruct people in future on what can be improved

Reply:
Sure :)

Reply:
agree